core/src/main/java/org/elasticsearch/plugins/MapperPlugin.java

```diff
@@ -52,4 +53,18 @@
     default Map<String, MetadataFieldMapper.TypeParser> getMetadataMappers() {
         return Collections.emptyMap();
     }
+
+    /**
+     * Returns a predicate that fields must match in order to be returned by get mappings, get index and get field mappings API.
+     * Useful to filter the fields that such API return. The predicate receives the index name and the field name as input arguments.
```

Comment:
Maybe add "and should return `true` if the to show the field and `false` to hide it".

Reply:
good idea